Build failure for ROS Iron: "Could not find a package configuration file provided by ament_cmake_flake8"

Reported at https://github.com/ros/rosdistro/pull/37815#issuecomment-1632069736 thanks to @Yadunund 

```
14:15:20 CMake Error at CMakeLists.txt:10 (find_package):
14:15:20   By not providing "Findament_cmake_flake8.cmake" in CMAKE_MODULE_PATH this
14:15:20   project has asked CMake to find a package configuration file provided by
14:15:20   "ament_cmake_flake8", but CMake did not find one.
14:15:20 
14:15:20   Could not find a package configuration file provided by
14:15:20   "ament_cmake_flake8" with any of the following names:
14:15:20 
14:15:20     ament_cmake_flake8Config.cmake
14:15:20     ament_cmake_flake8-config.cmake
14:15:20 
14:15:20   Add the installation prefix of "ament_cmake_flake8" to CMAKE_PREFIX_PATH or
14:15:20   set "ament_cmake_flake8_DIR" to a directory containing one of the above
14:15:20   files.  If "ament_cmake_flake8" provides a separate development package or
14:15:20   SDK, be sure it has been installed.
```

I confirmed the issue by removing the package in question then build (in this example `rocker` isn't needed at all, it can be a pure `docker run` but you get the idea).

<details><summary>Reproducible footage</summary>

```
export LINUX_DISTRO=jammy
export ROS_DISTRO=iron
export DOCKERIMG=ros:iron-ros-base
export PATH_LOCAL_WS=/home/noodle/ros/executive_smach

rocker --nvidia --x11 --network host  \
  --volume $PATH_LOCAL_WS/ros/executive_smach:/cws/src/executive_smach \
  --volume /dev:/dev  \
  --  \
  $DOCKERIMG bash
```
Now in the Docker container
```
export DIR_CWS=/cws && cd $DIR_CWS
source /opt/ros/iron/setup.bash
apt update && rosdep update && rosdep install --from-paths src --ignore-src -r -y

apt purge ros-iron-ament-cmake-flake8 && apt autoremove

colcon build
Starting >>> smach   
Starting >>> smach_msgs
/usr/lib/python3/dist-packages/setuptools/dist.py:723: UserWarning: Usage of dash-separated 'script-dir' will not be supported in future versions. Please use the underscore name 'script_dir' instead
  warnings.warn(
/usr/lib/python3/dist-packages/setuptools/dist.py:723: UserWarning: Usage of dash-separated 'install-scripts' will not be supported in future versions. Please use the underscore name 'install_scripts' instead
  warnings.warn(
--- stderr: smach                                                         
/usr/lib/python3/dist-packages/setuptools/dist.py:723: UserWarning: Usage of dash-separated 'script-dir' will not be supported in future versions. Please use the underscore name 'script_dir' instead
  warnings.warn(
/usr/lib/python3/dist-packages/setuptools/dist.py:157: UserWarning: newlines not allowed and will break in the future
  warnings.warn("newlines not allowed and will break in the future")
/usr/lib/python3/dist-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
  warnings.warn(
/cws/install/smach/lib/python3.10/site-packages/smach/state_machine.py:77: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  return {k:v for (k, v) in self.__dict__.items() if k is not "_state_transitioning_lock"}
---
Finished <<< smach [0.42s]
--- stderr: smach_msgs
CMake Error at CMakeLists.txt:10 (find_package):
  By not providing "Findament_cmake_flake8.cmake" in CMAKE_MODULE_PATH this
  project has asked CMake to find a package configuration file provided by
  "ament_cmake_flake8", but CMake did not find one.

  Could not find a package configuration file provided by
  "ament_cmake_flake8" with any of the following names:

    ament_cmake_flake8Config.cmake
    ament_cmake_flake8-config.cmake

  Add the installation prefix of "ament_cmake_flake8" to CMAKE_PREFIX_PATH or
  set "ament_cmake_flake8_DIR" to a directory containing one of the above
  files.  If "ament_cmake_flake8" provides a separate development package or
  SDK, be sure it has been installed.

---
Failed   <<< smach_msgs [0.45s, exited with code 1]

Summary: 1 package finished [0.55s]
  1 package failed: smach_msgs
  2 packages had stderr output: smach smach_msgs
  2 packages not processed
```
</details>

> I confirmed the issue by removing the package in question then build (in this example `rocker` isn't needed at all, it can be a pure `docker run` but you get the idea).

Just a follow-up comment to how I verified the issue https://github.com/ros/executive_smach/issues/111#issuecomment-1636453173, it looks like purging `ament_cmake_flake8` execerbate ros2' build pipeline unnecessarilly. Even with the suggested fix https://github.com/ros/executive_smach/pull/112, `colcon build still fails` shedding the same error.

```
:
Starting >>> smach_msgs
/usr/lib/python3/dist-packages/setuptools/dist.py:723: UserWarning: Usage of dash-separated 'script-dir' will not be supported in future versions. Please use the underscore name 'script_dir' instead
  warnings.warn(
/usr/lib/python3/dist-packages/setuptools/dist.py:723: UserWarning: Usage of dash-separated 'install-scripts' will not be supported in future versions. Please use the underscore name 'install_scripts' instead
  warnings.warn(
--- stderr: smach                                                         
/usr/lib/python3/dist-packages/setuptools/dist.py:723: UserWarning: Usage of dash-separated 'script-dir' will not be supported in future versions. Please use the underscore name 'script_dir' instead
  warnings.warn(
/usr/lib/python3/dist-packages/setuptools/dist.py:157: UserWarning: newlines not allowed and will break in the future
  warnings.warn("newlines not allowed and will break in the future")
/usr/lib/python3/dist-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
  warnings.warn(
/cws/install/smach/lib/python3.10/site-packages/smach/state_machine.py:77: SyntaxWarning: "is not" with a literal. Did you mean "!="?
  return {k:v for (k, v) in self.__dict__.items() if k is not "_state_transitioning_lock"}
---
Finished <<< smach [0.41s]
--- stderr: smach_msgs                         
CMake Error at /opt/ros/iron/share/rosidl_generator_py/cmake/rosidl_generator_py_generate_interfaces.cmake:335 (find_package):
  By not providing "Findament_cmake_flake8.cmake" in CMAKE_MODULE_PATH this
  project has asked CMake to find a package configuration file provided by
  "ament_cmake_flake8", but CMake did not find one.

  Could not find a package configuration file provided by
  "ament_cmake_flake8" with any of the following names:

    ament_cmake_flake8Config.cmake
    ament_cmake_flake8-config.cmake

  Add the installation prefix of "ament_cmake_flake8" to CMAKE_PREFIX_PATH or
  set "ament_cmake_flake8_DIR" to a directory containing one of the above
  files.  If "ament_cmake_flake8" provides a separate development package or
  SDK, be sure it has been installed.
Call Stack (most recent call first):
  /opt/ros/iron/share/ament_cmake_core/cmake/core/ament_execute_extensions.cmake:48 (include)
  /opt/ros/iron/share/rosidl_cmake/cmake/rosidl_generate_interfaces.cmake:316 (ament_execute_extensions)
  CMakeLists.txt:15 (rosidl_generate_interfaces)


---
Failed   <<< smach_msgs [1.04s, exited with code 1]

Summary: 1 package finished [1.14s]
  1 package failed: smach_msgs
  2 packages had stderr output: smach smach_msgs
  2 packages not processed
```

My guess is that somewhere in `rosidl_generate_interfaces` pipeline the `ament_flake8` pkg is depended?

--- a/executive_smach/package.xml
+++ b/executive_smach/package.xml
@@ -7,7 +7,7 @@
     This metapackage depends on the SMACH library and ROS SMACH integration
     packages.
   </description>
-  <maintainer email="gm130s@gmail.com">Isaac I. Y. Saito</maintainer>
+  <maintainer email="iisaac.saito@gmail.com">Isaac Saito</maintainer>
   <license>BSD</license>
 
   <url>http://ros.org/wiki/smach</url>

--- a/smach/package.xml
+++ b/smach/package.xml
@@ -11,7 +11,7 @@
     maintainable and modular code.
   </description>
 
-  <maintainer email="gm130s@gmail.com">Isaac I. Y. Saito</maintainer>
+  <maintainer email="iisaac.saito@gmail.com">Isaac Saito</maintainer>
   <license>BSD</license>
 
   <author>Jonathan Bohren</author>

--- a/smach_msgs/CMakeLists.txt
+++ b/smach_msgs/CMakeLists.txt
@@ -3,9 +3,7 @@ cmake_minimum_required(VERSION 3.22)
 project(smach_msgs)
 
 set(AMENT_DEPS
-  ament_cmake
-  ament_cmake_flake8
-  ament_cmake_pep257)
+  ament_cmake)
 foreach(dep IN ITEMS ${AMENT_DEPS})
   find_package(${dep} REQUIRED)
 endforeach()

--- a/smach_msgs/package.xml
+++ b/smach_msgs/package.xml
@@ -8,7 +8,7 @@
     interfaces for smach.
   </description>
 
-  <maintainer email="gm130s@gmail.com">Isaac I. Y. Saito</maintainer>
+  <maintainer email="iisaac.saito@gmail.com">Isaac Saito</maintainer>
   <license>BSD</license>
 
   <author>Jonathan Bohren</author>

--- a/smach_msgs/package.xml
+++ b/smach_msgs/package.xml
@@ -16,6 +16,9 @@
 
   <buildtool_depend>ament_cmake</buildtool_depend>
   <buildtool_depend>ament_cmake_cppcheck</buildtool_depend>
+  <!-- ament_cmake_{flake8, pep257} are added to workaround https://github.com/ros2/rosidl_python/issues/198 -->
+  <buildtool_depend>ament_cmake_flake8</buildtool_depend>
+  <buildtool_depend>ament_cmake_pep257</buildtool_depend>  
   <buildtool_depend>rosidl_default_generators</buildtool_depend>
 
   <build_depend>builtin_interfaces</build_depend>

--- a/smach_msgs/package.xml
+++ b/smach_msgs/package.xml
@@ -26,8 +26,6 @@
   <exec_depend>std_msgs</exec_depend>
 
   <test_depend>ament_cmake_cpplint</test_depend>
-  <test_depend>ament_cmake_flake8</test_depend>
-  <test_depend>ament_cmake_pep257</test_depend>
   <test_depend>ament_cmake_uncrustify</test_depend>
   <test_depend>ament_lint</test_depend>
 

